src/server/handlers/leave.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+//! Create subnet handler and parameters
+
+use crate::manager::SubnetManager;
+use crate::server::handlers::subnet::SubnetManagerPool;
+use crate::server::JsonRPCRequestHandler;
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fvm_shared::address::Address;
+use ipc_sdk::subnet_id::SubnetID;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use std::sync::Arc;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LeaveSubnetParams {
+    pub subnet: String,
+    pub from: Option<String>,
+}
+
+/// The create subnet json rpc method handler.
+pub(crate) struct LeaveSubnetHandler {
+    pool: Arc<SubnetManagerPool>,
+}
+
+impl LeaveSubnetHandler {
+    pub(crate) fn new(pool: Arc<SubnetManagerPool>) -> Self {
+        Self { pool }
+    }
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for LeaveSubnetHandler {
+    type Request = LeaveSubnetParams;
+    type Response = ();
+
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let conn = match self.pool.get(&request.subnet) {
```

Comment:
Same as above, you should get the conn for the parent of the subnet here.

Reply:
updated!